File: frontend/src/types/operations.ts
export type OperationalOrderStatus =
  | "new"
  | "pending_confirmation"
  | "confirmed"
  | "in_preparation"
  | "out_for_delivery"
  | "ready_for_pickup"
  | "completed"
  | "cancelled";

export type OrderItem = {
  id: number;
  product_name: string;
  quantity: number;
  unit_price: number;
  total_price: number;
  addons_json: Array<Record<string, unknown>> | null;
  notes: string | null;
};

export type OrderPrintJob = {
  id: number;
  trigger_status: string;
  printer_target: string | null;
  payload_text: string | null;
  printed: boolean;
  printed_at: string | null;
  created_at: string;
};

export type OperationalOrder = {
  id: number;
  company_id: number;
  conversation_id: number | null;
  code: string;
  status: OperationalOrderStatus;
  fulfillment_type: string;
  payment_method: string | null;
  customer_name: string | null;
  customer_phone: string | null;
  delivery_address: string | null;
  neighborhood: string | null;
  notes: string | null;
  subtotal: number;
  delivery_fee: number;
  discount_amount: number;
  total_amount: number;
  printed_at: string | null;
  created_at: string;
  updated_at: string;
  items: OrderItem[];
  print_jobs: OrderPrintJob[];
};

export type DashboardStat = {
  label: string;
  value: string | number;
  hint: string;
  tone: string;
};

export type ClientPanelData = {
  stats: DashboardStat[];
  orders: OperationalOrder[];
  inbox_counts: Record<string, number>;
  catalog_summary: Record<string, number>;
  inventory_alerts: Array<Record<string, string | number | boolean>>;
  finance_summary: {
    period_start: string;
    period_end: string;
    total_sold: number;
    average_ticket: number;
    orders_count: number;
    payment_breakdown: Record<string, number>;
    top_products: Array<{ name: string; quantity: number }>;
  };
  business_snapshot: Record<string, unknown>;
  ai_context_sources: Array<{ source: string; items: number; description: string }>;
};

export type DevPanelData = {
  global_stats: DashboardStat[];
  company_breakdown: Array<Record<string, string | number | boolean>>;
  plan_breakdown: Array<{ plan: string; companies: number }>;
  ai_usage: {
    total_tokens: number;
    estimated_cost: number;
    models: Record<string, number>;
  };
  global_logs: Array<{ title: string; level: string; description: string }>;
  channel_summary: Array<Record<string, string | number | boolean | null>>;
};

export type InventoryItem = {
  id: number;
  company_id: number;
  product_id: number | null;
  name: string;
  unit: string;
  current_quantity: number;
  low_stock_threshold: number;
  active: boolean;
  available_for_sale: boolean;
  created_at: string;
  updated_at: string;
};

export type FinanceSummary = ClientPanelData["finance_summary"];

File: app/schemas/operations.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class OrderItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_name: str
    quantity: int
    unit_price: float
    total_price: float
    addons_json: list[dict] | None = None
    notes: str | None = None


class OrderPrintJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    trigger_status: str
    printer_target: str | None
    payload_text: str | None
    printed: bool
    printed_at: datetime | None
    created_at: datetime


class OrderResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    company_id: int
    conversation_id: int | None
    code: str
    status: str
    fulfillment_type: str
    payment_method: str | None
    customer_name: str | None
    customer_phone: str | None
    delivery_address: str | None
    neighborhood: str | None
    notes: str | None
    subtotal: float
    delivery_fee: float
    discount_amount: float
    total_amount: float
    printed_at: datetime | None
    created_at: datetime
    updated_at: datetime
    items: list[OrderItemResponse] = Field(default_factory=list)
    print_jobs: list[OrderPrintJobResponse] = Field(default_factory=list)


class OrderStatusUpdate(BaseModel):
    status: str = Field(pattern="^(new|pending_confirmation|confirmed|in_preparation|out_for_delivery|ready_for_pickup|completed|cancelled)$")


class ClientStatItem(BaseModel):
    label: str
    value: str | int | float
    hint: str
    tone: str = "default"


class ClientDashboardResponse(BaseModel):
    stats: list[ClientStatItem]
    orders: list[OrderResponse]
    inbox_counts: dict[str, int]
    catalog_summary: dict[str, int]
    inventory_alerts: list[dict]
    finance_summary: dict
    business_snapshot: dict
    ai_context_sources: list[dict]


class DevDashboardResponse(BaseModel):
    global_stats: list[ClientStatItem]
    company_breakdown: list[dict]
    plan_breakdown: list[dict]
    ai_usage: dict
    global_logs: list[dict]
    channel_summary: list[dict]


class InventoryItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    company_id: int
    product_id: int | None
    name: str
    unit: str
    current_quantity: float
    low_stock_threshold: float
    active: bool
    available_for_sale: bool
    created_at: datetime
    updated_at: datetime


class FinanceSummaryResponse(BaseModel):
    period_start: date
    period_end: date
    total_sold: float
    average_ticket: float
    orders_count: int
    payment_breakdown: dict[str, float]
    top_products: list[dict]
